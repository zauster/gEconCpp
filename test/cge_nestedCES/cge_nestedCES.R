# ###########################################################################
# (c) Chancellery of the Prime Minister 2012-2015                            
#                                                                            
# Authors: Marta Retkiewicz                                                  
# ###########################################################################
# Simple CGE model with Cobb-Douglas production function
# ###########################################################################

# load gEcon package
library(gEcon)

# make and load the model
cge.ces <- make_model("cge_calibration_ces.gcn")

# set model parameters' values
free_par <- c(omega          = 2,
              pi_h__2          = 0.5,
              d_data__B__1     = 11.700,
              d_data__B__2     = 30.790,
              d_data__C__1     = 18.600,
              d_data__C__2     = 43.600,
              ls_data__1       = 18.170,
              ls_data__2       = 70.070,
              ks_data__1       = 65.070,
              ks_data__2       = 68.770,
              l_data__A        = 9.440,
              l_data__B        = 31.600,
              l_data__C        = 47.200,
              x_data__A__A     = 68.400,
              x_data__A__B     = 131.010,
              x_data__A__C     = 28.280,
              x_data__B__A     = 111.910,
              x_data__B__B     = 92.300,
              x_data__B__C     = 86.920,
              x_data__C__A     = 117.230,
              x_data__C__B     = 43.700,
              x_data__C__C     = 111.650,
              y_data__A        = 345.080,
              y_data__B        = 333.620,
              y_data__C        = 334.780)
              
cge.ces <- set_free_par(cge.ces, free_par = free_par)

# set calibrated parameters' initial values
cge.ces <- initval_calibr_par(cge.ces, 
                             c(beta_x__A__A   = 1/5,
                               beta_x__A__B     = 1/5,
                               beta_x__A__C     = 1/5,
                               beta_x__B__A     = 1/5,
                               beta_x__B__B     = 1/5,
                               beta_x__B__C     = 1/5,
                               beta_x__C__A     = 1/5,
                               beta_x__C__B     = 1/5,
                               beta_x__C__C     = 1/5,
                               beta_k__A        = 1/5,
                               beta_k__B        = 1/5,
                               beta_k__C        = 1/5,
                               beta_l__A        = 1/5,
                               beta_l__B        = 1/5,
                               beta_l__C        = 1/5,
                               gamma__A         = 4,
                               gamma__B         = 4,
                               gamma__C         = 4))

# find and print equilibrium values
cge.ces <- steady_state(cge.ces)
get_ss_values(cge.ces, to_tex = TRUE)

# print calibrated parameters' values
calibr_par <- setdiff(get_par_names(cge.ces), names(free_par))
get_par_values(cge.ces, parameters = calibr_par, to_tex = TRUE)

