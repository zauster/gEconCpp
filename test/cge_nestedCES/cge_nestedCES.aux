\relax 
\providecommand\hyper@newdestlabel[2]{}
\providecommand\HyperFirstAtBeginDocument{\AtBeginDocument}
\HyperFirstAtBeginDocument{\ifx\hyper@anchor\@undefined
\global\let\oldcontentsline\contentsline
\gdef\contentsline#1#2#3#4{\oldcontentsline{#1}{#2}{#3}}
\global\let\oldnewlabel\newlabel
\gdef\newlabel#1#2{\newlabelxx{#1}#2}
\gdef\newlabelxx#1#2#3#4#5#6{\oldnewlabel{#1}{{#2}{#3}}}
\AtEndDocument{\ifx\hyper@anchor\@undefined
\let\contentsline\oldcontentsline
\let\newlabel\oldnewlabel
\fi}
\fi}
\global\let\hyper@last\relax 
\gdef\HyperFirstAtBeginDocument#1{#1}
\providecommand\HyField@AuxAddToFields[1]{}
\providecommand\HyField@AuxAddToCoFields[2]{}
\@writefile{toc}{\contentsline {section}{\numberline {1}CONSUMER $i\in {C\tmspace  -\thinmuskip {.1667em}O\tmspace  -\thinmuskip {.1667em}U\tmspace  -\thinmuskip {.1667em}N\tmspace  -\thinmuskip {.1667em}T\tmspace  -\thinmuskip {.1667em}R\tmspace  -\thinmuskip {.1667em}I\tmspace  -\thinmuskip {.1667em}E\tmspace  -\thinmuskip {.1667em}S}$ $h\in {C\tmspace  -\thinmuskip {.1667em}O\tmspace  -\thinmuskip {.1667em}N\tmspace  -\thinmuskip {.1667em}S\tmspace  -\thinmuskip {.1667em}U\tmspace  -\thinmuskip {.1667em}M\tmspace  -\thinmuskip {.1667em}E\tmspace  -\thinmuskip {.1667em}R\tmspace  -\thinmuskip {.1667em}S}$}{1}{section.1}}
\@writefile{toc}{\contentsline {subsection}{\numberline {1.1}Optimisation problem}{1}{subsection.1.1}}
\@writefile{toc}{\contentsline {subsection}{\numberline {1.2}Identities}{1}{subsection.1.2}}
\@writefile{toc}{\contentsline {subsection}{\numberline {1.3}First order conditions}{1}{subsection.1.3}}
\@writefile{toc}{\contentsline {section}{\numberline {2}FIRM $i\in {C\tmspace  -\thinmuskip {.1667em}O\tmspace  -\thinmuskip {.1667em}U\tmspace  -\thinmuskip {.1667em}N\tmspace  -\thinmuskip {.1667em}T\tmspace  -\thinmuskip {.1667em}R\tmspace  -\thinmuskip {.1667em}I\tmspace  -\thinmuskip {.1667em}E\tmspace  -\thinmuskip {.1667em}S}$ $s\in {S\tmspace  -\thinmuskip {.1667em}E\tmspace  -\thinmuskip {.1667em}C\tmspace  -\thinmuskip {.1667em}T\tmspace  -\thinmuskip {.1667em}O\tmspace  -\thinmuskip {.1667em}R\tmspace  -\thinmuskip {.1667em}S}$}{2}{section.2}}
\@writefile{toc}{\contentsline {subsection}{\numberline {2.1}Optimisation problem}{2}{subsection.2.1}}
\@writefile{toc}{\contentsline {subsection}{\numberline {2.2}First order conditions}{2}{subsection.2.2}}
\@writefile{toc}{\contentsline {subsection}{\numberline {2.3}First order conditions after reduction}{2}{subsection.2.3}}
\@writefile{toc}{\contentsline {section}{\numberline {3}EQUILIBRIUM}{3}{section.3}}
\@writefile{toc}{\contentsline {subsection}{\numberline {3.1}Identities}{3}{subsection.3.1}}
\@writefile{toc}{\contentsline {section}{\numberline {4}Equilibrium relationships (before expansion and reduction)}{3}{section.4}}
\@writefile{toc}{\contentsline {section}{\numberline {5}Equilibrium relationships (after expansion and reduction)}{4}{section.5}}
\@writefile{toc}{\contentsline {section}{\numberline {6}Calibrating equations}{19}{section.6}}
