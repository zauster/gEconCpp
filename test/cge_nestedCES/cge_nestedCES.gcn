# ###########################################################################
# Authors: Oliver Reiter
# ###########################################################################
# (Multi-Country) CGE model with CES production function
# ###########################################################################


options
{
    output R = TRUE;
    output LaTeX = TRUE;
    output LaTeX landscape = TRUE;
    output logfile = TRUE;
};

indexsets
{
    SECTORS = { 'A' .. 'G' };
    CONSUMERS = { '1' .. '3' };
    ## COUNTRIES = { 'AUT' };
    COUNTRIES = { 'AUT' , 'DEU', 'FRA', 'GBR', 'SVN', 'USA'};
};

block <i::COUNTRIES> <h::CONSUMERS> CONSUMER
{
    definitions
    {
        u<i, h>[] = (SUM<s::SECTORS>(alpha<i, s, h> * D<i, s, h>[]^((omega - 1)/omega)))^(omega/(omega - 1));
    };
    controls
    {
        <s::SECTORS>D<i, s, h>[]; # Household h demand for good s
    };
    objective
    {
        U<i, h>[] = u<i, h>[];
    };
    constraints
    {
        INC<i, h>[] + PI<i, h>[] = SUM<s::SECTORS>(D<i, s, h>[] * p<i, s>[]);
    };
    identities
    {
        INC<i, h>[] = K<i, h>[] * p_k[] + L<i, h>[];
        K<i, h>[] = ks_data<i, h>;
        L<i, h>[] = ls_data<i, h>;
    };
    calibration
    {
        <s::SECTORS\'A'>D<i, s, h>[] = d_data<i, s, h> -> alpha<i, s, h>;
        SUM<s::SECTORS>(alpha<i, s, h>^omega) = 1 -> alpha<i, 'A', h>;
    };
};



block <i::COUNTRIES> <s::SECTORS> FIRM
{
    controls
    {
        Y<i, s>[], K<i, s>[], L<i, s>[], <si::SECTORS>X<i, si, s>[];
    };
    objective
    {
        pi<i, s>[] = Y<i, s>[] * p<i, s>[] - (K<i, s>[] * p_k[] + L<i, s>[]) - SUM<si::SECTORS>(X<i, si, s>[] * p<i, si>[]);
    };
    constraints
    {
        ## original Cobb-Douglas
        ## Y<i, s>[] = gamma<i, s> * K<i, s>[]^beta_k<i, s> * L<i, s>[]^beta_l<i, s> * PROD<si::SECTORS>(X<i, si, s>[]^beta_x<i, si, s>);

        ## CES function over all factors of production
        ## Y<i, s>[] = gamma<i, s> * (beta_k<i, s> * K<i, s>[]^rho + beta_l<i, s> * L<i, s>[]^rho + SUM<si::SECTORS>(beta_x<i, si, s> * X<i, si, s>[]^rho))^(rts / rho);

        ## Cobb-Douglas with nested CES function over all factors of
        ## production
        Y<i, s>[] = gamma<i, s> * (K<i, s>[]^eta_k<i, s> * L<i, s>[]^eta_l<i, s> *
        (SUM<si::SECTORS>(beta_x<i, si, s> * X<i, si, s>[]^rho<i>))^(eta_x<i, s> / rho<i>));
        
    };
    calibration
    {
        ## CD and CES:
        ## beta_k<i, s> + beta_l<i, s> + SUM<si::SECTORS>(beta_x<si, s>) = 1 -> beta_k<i, s>;
        ## L<i, s>[] = l_data<i, s> -> beta_l<i, s>;
        ## <si::SECTORS>X<i, si, s>[] = x_data<i, si, s> -> beta_x<i, si, s>;
        ## Y<i, s>[] = y_data<i, s> -> gamma<i, s>;
        ## rts = 1;
        ## rho = 2;

        ## CD and nested CES:
        ## X<i, s, s>[] = x_data<i, s, s>;
        <si::SECTORS\s>X<i, si, s>[] = x_data<i, si, s> -> beta_x<i, si, s>;
        SUM<si::SECTORS>(beta_x<i, si, s>) = 1 -> beta_x<i, s, s>;
        
        K<i, s>[] = k_data<i, s> -> eta_k<i, s>;
        L<i, s>[] = l_data<i, s> -> eta_l<i, s>;
        eta_k<i, s> + eta_l<i, s> + eta_x<i, s> = 1 -> eta_x<i, s>;

        Y<i, s>[] = y_data<i, s> -> gamma<i, s>;
        ## rho<i> = 2;
    };
};

block EQUILIBRIUM
{
    identities
    {
        <i::COUNTRIES> SUM<h::CONSUMERS>(K<i, h>[]) = SUM<s::SECTORS>(K<i, s>[]);
        <s::SECTORS>p<'AUT', s>[] = 1;
        <i::COUNTRIES\'AUT'> <s::SECTORS\'A'>p<i, s>[] = 1;
        <i::COUNTRIES> <h::CONSUMERS>PI<i, h>[] = SUM<s::SECTORS>(pi<i, s>[]) * pi_h<i, h>;
    };
    calibration
    {
        <i::COUNTRIES> SUM<h::CONSUMERS>(pi_h<i, h>) = 1 -> pi_h<i, '1'>;
    }
};
