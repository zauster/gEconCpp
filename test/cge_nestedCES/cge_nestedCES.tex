% Generated on 2017-04-17 22:20:48 by gEcon ver. 1.0.2 (2016-12-05)
% http://gecon.r-forge.r-project.org/

% Model name: cge_nestedCES

\documentclass[10pt,a4paper]{article}
\usepackage[utf8]{inputenc}
\usepackage{color}
\usepackage{graphicx}
\usepackage{epstopdf}
\usepackage{hyperref}
\usepackage{amsmath}
\usepackage{amssymb}
\usepackage{pdflscape}
\numberwithin{equation}{section}
\usepackage[top = 2.5cm, bottom=2.5cm, left = 2.0cm, right=2.0cm]{geometry}
\begin{document}

\begin{landscape}
\begin{flushleft}{\large
Generated  on 2017-04-17 22:20:48 by \href{http://gecon.r-forge.r-project.org/}{\texttt{gEcon}} version 1.0.2 (2016-12-05)\\
Model name: \verb+cge_nestedCES+
}\end{flushleft}

\input{cge_nestedCES.model.tex}
\end{landscape}
\input{cge_nestedCES.results.tex}

\end{document}

