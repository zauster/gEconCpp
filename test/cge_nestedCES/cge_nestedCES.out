\BOOKMARK [1][-]{section.1}{CONSUMER iCOUNTRIES hCONSUMERS}{}% 1
\BOOKMARK [2][-]{subsection.1.1}{Optimisation problem}{section.1}% 2
\BOOKMARK [2][-]{subsection.1.2}{Identities}{section.1}% 3
\BOOKMARK [2][-]{subsection.1.3}{First order conditions}{section.1}% 4
\BOOKMARK [1][-]{section.2}{FIRM iCOUNTRIES sSECTORS}{}% 5
\BOOKMARK [2][-]{subsection.2.1}{Optimisation problem}{section.2}% 6
\BOOKMARK [2][-]{subsection.2.2}{First order conditions}{section.2}% 7
\BOOKMARK [2][-]{subsection.2.3}{First order conditions after reduction}{section.2}% 8
\BOOKMARK [1][-]{section.3}{EQUILIBRIUM}{}% 9
\BOOKMARK [2][-]{subsection.3.1}{Identities}{section.3}% 10
\BOOKMARK [1][-]{section.4}{Equilibrium relationships \(before expansion and reduction\)}{}% 11
\BOOKMARK [1][-]{section.5}{Equilibrium relationships \(after expansion and reduction\)}{}% 12
\BOOKMARK [1][-]{section.6}{Calibrating equations}{}% 13
